src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -4613,7 +4613,42 @@ public Response getCanDownloadAtLeastOneFile(@Context ContainerRequestContext cr
             return ok(permissionService.canDownloadAtLeastOneFile(req, datasetVersion));
         }, getRequestUser(crc));
     }
-    
+
+    @PUT
+    @AuthRequired
+    @Path("{identifier}/pidReconcile/{pididentifier}")
+    public Response reconcilePid(@Context ContainerRequestContext crc, @PathParam("identifier") String datasetId,
+                                 @PathParam("pididentifier") String generatorId) throws WrappedResponse {
```

Comment:
I'd suggest not having this param and always using the effectivePidGenerator. That can be set at the dataset level, so you could pick any PidProvider by setting it for the dataset first, but the more usual situation would be that the effectivePidGenerator is the one set for the collection, so you'd automatically get the provider for the new collection after a move.

Reply:
That's a good idea!